Source/WebCore/page/PointerLockController.cpp

```diff
@@ -40,44 +42,83 @@
 
 namespace WebCore {
 
+class UnadjustedMovementPlatformMouseEvent : public PlatformMouseEvent {
+public:
+    UnadjustedMovementPlatformMouseEvent(const PlatformMouseEvent &src)
+        : PlatformMouseEvent(src.position(), src.globalPosition(), src.button(), src.type(), src.clickCount(), src.modifiers(), src.timestamp(), src.force(), src.syntheticClickType(), src.pointerId())
+    {
+        m_movementDelta = m_unadjustedMovementDelta = src.unadjustedMovementDelta();
+    }
+};
+
 PointerLockController::PointerLockController(Page& page)
     : m_page(page)
 {
 }
 
-void PointerLockController::requestPointerLock(Element* target)
+PointerLockController::~PointerLockController()
+{
+}
+
+void PointerLockController::requestPointerLock(Element* target, bool unadjustedMovement, RefPtr<DeferredPromise> promise)
 {
     if (!target || !target->isConnected() || m_documentOfRemovedElementWhileWaitingForUnlock) {
         enqueueEvent(eventNames().pointerlockerrorEvent, target);
+        if (promise)
+            promise->reject(WrongDocumentError);
         return;
     }
 
     if (m_documentAllowedToRelockWithoutUserGesture != &target->document() && !UserGestureIndicator::processingUserGesture()) {
         enqueueEvent(eventNames().pointerlockerrorEvent, target);
+        // If the request was not started from an engagement gesture and the Document has not previously released a successful Pointer Lock with exitPointerLock():
```

Comment:
Nit: but the comment just describes the previous line of code and doesn't give any special insight. Remove if you agree? No hard preference

Reply:
These comments I copy and pasted verbatim from the updated spec. It's actually just a coincidence that it describes the code. My goal was to make it clear how each case in the spec corresponded to the code. I can eliminate if you like, but I found it helpful to keep my own thinking straight when I wrote the code, and I hope that it will be useful to somebody who might come along after me.